mergefs.go

```diff
@@ -25,3 +28,64 @@ func (mfs MergedFS) Open(name string) (fs.File, error) {
 	}
 	return nil, os.ErrNotExist
 }
+
+// ReadDir reads from the directory, and produces a DirEntry array of different
+// directories.
+//
+// It iterates through all different filesystems that exist in the mfs MergeFS
+// filesystem slice and it identifies overlapping directories that exist in different
+// filesystems
+func (mfs MergedFS) ReadDir(name string) ([]fs.DirEntry, error) {
+	dirsMap := make(map[string]fs.DirEntry)
+	for _, filesystem := range mfs.filesystems {
+		if fsys, ok := filesystem.(fs.ReadDirFS); ok {
+			dir, err := fsys.ReadDir(name)
+			if err != nil {
+				if errors.Is(err, fs.ErrNotExist) {
+					log.Debug().Msgf("directory in filepath %s was not found in filesystem", name)
+					continue
+				}
+				return nil, err
+			}
+			for _, v := range dir {
+				if _, ok := dirsMap[v.Name()]; !ok {
+					dirsMap[v.Name()] = v
+				}
+			}
+			continue
+		}
+
+		file, err := filesystem.Open(name)
+		if err != nil {
+			log.Debug().Msgf("filepath %s was not found in filesystem", name)
+			continue
+		}
+
+		dir, ok := file.(fs.ReadDirFile)
+		if !ok {
+			return nil, &fs.PathError{Op: "readdir", Path: name, Err: errors.New("not implemented")}
+		}
+
```

Comment:
Better to make sure the file gets closed:

```suggestion
defer file.Close()
```

Reply:
(But this whole part would be unnecessary with fs.ReadDir)

Reply:
The file is closed on line 81 - wouldn't go and `defer` inside a for loop